src/planner/planner.c

```diff
@@ -1310,30 +1344,35 @@ timescaledb_get_relation_info_hook(PlannerInfo *root, Oid relation_objectid, boo
 		}
 		case TS_REL_CHUNK_STANDALONE:
 		case TS_REL_CHUNK_CHILD:
-		{
 			ts_create_private_reloptinfo(rel);
 
-			if (ts_guc_enable_transparent_decompression && TS_HYPERTABLE_HAS_COMPRESSION_TABLE(ht))
+			/*
+			 * To save time, don't plan index scans for empty uncompressed
+			 * tables of fully compressed chunks.
+			 *
+			 * Normally this is done in ts_set_append_rel_pathlist() using the
+			 * Chunk data cached by our hypertable expansion, but it cases when
+			 * these functions don't run, we have to do it here.
+			 */
+			const bool use_transparent_decompression =
+				ts_guc_enable_transparent_decompression && TS_HYPERTABLE_HAS_COMPRESSION_TABLE(ht);
+			const bool is_standalone_chunk = (type == TS_REL_CHUNK_STANDALONE) &&
+											 !TS_HYPERTABLE_IS_INTERNAL_COMPRESSION_TABLE(ht);
+			const bool is_child_chunk_in_update =
+				(type == TS_REL_CHUNK_CHILD) && IS_UPDL_CMD(query);
+			if (use_transparent_decompression && (is_standalone_chunk || is_child_chunk_in_update))
 			{
-				RangeTblEntry *chunk_rte = planner_rt_fetch(rel->relid, root);
-				Chunk *chunk = ts_chunk_get_by_relid(chunk_rte->relid, true);
-
-				if (chunk->fd.compressed_chunk_id != INVALID_CHUNK_ID)
+				TimescaleDBPrivate *fdw_private = (TimescaleDBPrivate *) rel->fdw_private;
+				Assert(fdw_private->cached_chunk_struct == NULL);
+				fdw_private->cached_chunk_struct =
+					ts_chunk_get_by_relid(rte->relid, /* fail_if_not_found = */ true);
+				if (!ts_chunk_is_partial(fdw_private->cached_chunk_struct) &&
+					ts_chunk_is_compressed(fdw_private->cached_chunk_struct))
 				{
-					/* Planning indexes is expensive, and if this is a fully compressed chunk, we
-					 * know we'll never need to use indexes on the uncompressed version, since
-					 * all the data is in the compressed chunk anyway. Therefore, it is much
-					 * faster if we simply trash the indexlist here and never plan any useless
-					 * IndexPaths at all.
-					 * If the chunk is partially compressed, then we should enable indexScan
```

Comment:
Could we keep this comment about the partially compressed chunks?

Reply:
This comment is above the `if` condition now, I extended it a little.